### ISSUE TITLE
[Request] Treat <C-[> as <Esc> across OniVim

### ISSUE THREAD

**Issue by LeonT-A** [onivim/oni2#2368]
Vim treats `<C-[>` as an alternative to pressing `<Esc>`. Once you train yourself to use this, it is much quicker and less disorientating than reaching for the `<Esc>` key. 

At the moment, OniVim doesn't acknowledge `<C-[>` either in Sneak-Mode or in the Command Pallet. It also un-indents lines in visual mode(s) instead of returning to Normal mode.

However, during insert mode, command mode, and search mode, it DOES respect the relationship to `<Esc>`. Therefore the behavior is inconsistent.

`:h <Esc>`
```
notation	      meaning	     equivalent	decimal         value(s)	
-----------------------------------------------------------------------
<Esc>	       escape		CTRL-[	 27	       *escape* *<Esc>*
```

`:h CTRL-[`
```
CTRL-[						c_CTRL-[ c_<Esc> c_Esc
<Esc>		When typed and 'x' not present in 'cpoptions', quit
		Command-line mode without executing.  In macros or when 'x'
		present in 'cpoptions', start entered command.
		Note: If your <Esc> key is hard to hit on your keyboard, train
		yourself to use CTRL-[.
```

**Comment by joshtwo** [comment]
I'm glad this is confirmed as a bug. I only just started using Onivim and I swore this was happening occasionally, but I could never reproduce it on purpose.

**Comment by bryphe** [comment]
This is fixed now in #2878 - `Ctrl-[` should be treated as `Esc` across the UI. Will close out as fixed by #2878, but please let me know if you hit this again anywhere
